mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMessageIdManager.java

```diff
@@ -296,11 +297,20 @@ private void addMessageToMailboxes(MailboxMessage mailboxMessage, Set<MailboxId>
 
         for (MailboxId mailboxId : mailboxIds) {
             SimpleMailboxMessage copy = SimpleMailboxMessage.copy(mailboxId, mailboxMessage);
+            copy.setFlags(sanitizedFlags(mailboxId, mailboxSession, mailboxMessage.createFlags()));
             save(mailboxSession, messageIdMapper, copy);
             dispatcher.added(mailboxSession, mailboxMapper.findMailboxById(mailboxId), copy);
         }
     }
 
+    private Flags sanitizedFlags(MailboxId mailboxId, MailboxSession mailboxSession, Flags flags) throws MailboxException {
```

Comment:
We should make sure argument order is consistent. Usually, session is the last argument, mailboxId is the first.